### PR TITLE
[KED-2615] Nudge user to upgrade Kedro-Viz from the CLI

### PR DESCRIPTION
**Pull request by limdauto**
## Description

This PR shows a warning to encourage user to upgrade Kedro-Viz if there is a newer version on PyPI.

## Development notes

Screenshot 2022-01-10 at 17 39 03

## QA notes

I have tested this manually

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,16 +24,13 @@ e2e-tests: build
...
-	mypy --config-file=package/mypy.ini package &
-	wait
-	echo "Done linting"
+	cd package && isort .
+	black package/kedro_viz package/tests package/features
+	pylint --rcfile=package/.pylintrc -j 0 package/kedro_viz
+	pylint --rcfile=package/.pylintrc -j 0 --disable=protected-access,missing-docstring,redefined-outer-name,no-self-use,invalid-name,too-few-public-methods,no-member,unused-argument,duplicate-code package/tests
+	pylint --rcfile=package/.pylintrc -j 0 --disable=missing-docstring,no-name-in-module,unused-argument package/features
+	flake8 --config=package/.flake8 package
+	mypy --config-file=package/mypy.ini package
```
Comment (limdauto) [inline Makefile:33]: I changed this in another PR to parallelise this step but I didn't know `&` and `wait` doesn't work on makefile across serveral lines, so I reverted it here.

**Comment by datajoely** [comment]
Do we want to make a note about Kedro-Viz being designed to be backward compatible?

File: package/kedro_viz/integrations/pypi/__init__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+"""`kedro_viz.integrations.pypi` provides interface to integrate Kedro-Viz with PyPI."""
```
Comment (AntonyMilneQB) [inline package/kedro_viz/integrations/pypi/__init__.py:1]:
```suggestion
"""`kedro_viz.integrations.pypi` provides an interface to integrate Kedro-Viz with PyPI."""
```

File: package/kedro_viz/launchers/cli.py [hunk 3]
```diff
@@ -71,6 +74,22 @@ def commands():
 )
 def viz(host, port, browser, load_file, save_file, pipeline, env, autoreload):
     """Visualise a Kedro pipeline using Kedro viz."""
+    current_version = version.parse(__version__)
+    latest_version = get_latest_version()
+
+    if latest_version is not None and current_version < latest_version:
+        click.echo(
+            click.style(
+                "WARNING: You are using an older version of Kedro Viz. "
```
Comment (AntonyMilneQB) [inline package/kedro_viz/launchers/cli.py:83]:
```suggestion
                "WARNING: You are using an old version of Kedro Viz. "
```

**Review by AntonyMilneQB — APPROVED** [review]
Nice change, and 💯 tests.

Just a couple of ideas - nothing big, so feel free to ignore:
* should we explain why a user might want to upgrade? "It's backwards-compatible so will work with your current project straight away and will let you use lots of cool new features"
* is there any way of making it even easier for someone to run the `pip install -U` command so they don't even need to copy/paste it? Like somehow making it clickable or so they just type "y" to execute it (this would be annoying though because you get all the kedro viz console messages after that). I think probably not, but just mentioning it in case you are struck with inspiration

**Review by MerelTheisenQB — APPROVED** [review]
Great change! 👍

**Review by rashidakanchwala — APPROVED** [review]
LGTM!

**Comment by limdauto** [comment]
@AntonyMilneQB @datajoely yea happy to highlight it if someone can help me with the wording. It's already quite long.

@AntonyMilneQB re autoupdate: yes, that's the next iteration. This is a quick fix first, then the idea is we do autoupdate like VSCode or even from the CLI. But that's more involved (need to launch and manage another process, etc.)

**Pull request merged**